crates/net/network/src/transactions/fetcher.rs

```diff
@@ -339,13 +343,18 @@ impl TransactionFetcher {
                 for peer_id in ended_sessions {
                     backups.remove(&peer_id);
                 }
-                backups.insert(peer_id);
+
+                // might receive hash in more than one announcement from same peer within time 
+                // frame it takes to successfully fetch it
+                if !backups.contains(&peer_id) {
+                    backups.insert(peer_id);
+                }
```

Comment:
why is this check required?

we insert if it doesn't exist, isn't this the same as just insert?

Reply:
true, lru doesn't allow duplicates, it promotes the entry if it already exists. then nvm, just over caution, will remove.